### ISSUE TITLE
Attribute error User instance has no attribute 'UserProfile' in function getAllUsers

### ISSUE THREAD

**Issue by jfthuong** [RallyTools/RallyRestToolkitForPython#57]
The following block of code returns an error when the user has no attribute 'UserProfile'.

restapi.py, lines 571 to 575:
```python
            # get any matching user profiles (aka mups), there really should only be 1 matching...
            mups = [prof for prof in profiles 
                          if hasattr(user, 'UserProfile') and prof._ref == user.UserProfile._ref] 
            if not mups:
                problem = "unable to find a matching UserProfile record for User: %s  UserProfile: %s"
                warning("%s" % (problem % (user.DisplayName, user.UserProfile)))
                continue
```

If the user has no attribute, 'UserProfile', the program enter in the condition "if no mups"; the line " warning(...)" raises then an error instead of a warning.

**Comment by jfthuong** [comment]
If the problem is confirmed, I might suggest the following fix:
```python
            if not mups:
                UserProfile = user.UserProfile if hasattr(user,'UserProfile') else "Unknown"
				problem = "unable to find a matching UserProfile record for User: %s  UserProfile: %s"
				warning("%s" % (problem % (user.DisplayName, UserProfile)))
                continue
```

**Comment by klehman-rally** [comment]
UserProfile issue root cause is that user credentials for connection are not in the SubscriptionAdmin role.  Other code implemented to address this in 1.2.0.
